client/packages/system/src/Stock/DetailView/DetailView.tsx

```diff
@@ -30,36 +34,54 @@ export const StockLineDetailView: React.FC = () => {
     isDirty,
     update: { update, isUpdating },
   } = useStockLine(id);
-  // const { dispatchEvent, addEventListener, removeEventListener } =
-  //   usePluginEvents();
+  const { dispatchEvent, addEventListener, removeEventListener } =
+    usePluginEvents();
   const {
     urlQuery: { tab },
   } = useUrlQuery();
   const { success, error } = useNotification();
   const t = useTranslation('inventory');
   const { setSuffix } = useBreadcrumbs();
 
+  const repackModalController = useToggle();
+  const adjustmentModalController = useToggle();
+
+  const [hasPluginChanged, setHasPluginChanged] = useState(false);
```

Comment:
So that "hasChanged" you asked about in the last PR (@lache-melvin ) -- turns out it was for the plugins. Looks like any plugins need to maintain their own "hasChanged" state in addition to the draft `isDirty`. I've renamed it to make it clearer what it belongs to.